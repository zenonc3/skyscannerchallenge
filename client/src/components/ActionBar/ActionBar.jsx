import React from 'react';
import BpkButton from 'bpk-component-button/src/BpkButton';
import BpkSmallPriceAlertIcon from 'bpk-component-icon/sm/price-alerts';
import { withButtonAlignment } from 'bpk-component-icon';
import STYLES from './ActionBar.scss';

const c = className => STYLES[className] || 'UNKNOWN';

const AlignedBpkSmallPriceAlertIcon = withButtonAlignment(BpkSmallPriceAlertIcon);

const ActionBar = () => (
  <div className={c('ActionBar__container')}>
    <BpkButton link className={c('ActionBar__filter')}>Filter</BpkButton>
    <BpkButton link className={c('ActionBar__sort')}>Sort</BpkButton>
    <div className={c('ActionBar__alert')}>
      <BpkButton link >
        <AlignedBpkSmallPriceAlertIcon /> Price Alerts
      </BpkButton>
    </div>

  </div>
);

export default ActionBar;
