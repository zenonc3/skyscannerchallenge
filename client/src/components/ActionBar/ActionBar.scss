@import '~bpk-mixins';

.ActionBar {
  &__container {
    display: flex;
    width: 100%;
    flex-flow: wrap row;
  }

  &__filter {
    display: flex;
    padding: $bpk-spacing-sm;
    flex-direction: column;
  }

  &__sort {
    display: flex;
    padding: $bpk-spacing-sm;
    flex-direction: column;
  }

  &__alert {
    display: flex;
    padding: $bpk-spacing-sm;
    flex-direction: column;
    justify-content: center;
    align-items: flex-end;
    flex: 1;
  }
}
