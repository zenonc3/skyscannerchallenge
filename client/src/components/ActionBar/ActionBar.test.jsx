import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ActionBar from './ActionBar';

describe('Pricing', () => {
  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<ActionBar />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

