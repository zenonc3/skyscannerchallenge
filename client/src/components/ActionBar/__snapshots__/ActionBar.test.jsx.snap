// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pricing should render correctly 1`] = `
<div
  className="ActionBar__container"
>
  <BpkButton
    blank={false}
    className="ActionBar__filter"
    destructive={false}
    disabled={false}
    featured={false}
    href={null}
    iconOnly={false}
    large={false}
    link={true}
    onClick={null}
    outline={false}
    rel={null}
    secondary={false}
    submit={false}
  >
    Filter
  </BpkButton>
  <BpkButton
    blank={false}
    className="ActionBar__sort"
    destructive={false}
    disabled={false}
    featured={false}
    href={null}
    iconOnly={false}
    large={false}
    link={true}
    onClick={null}
    outline={false}
    rel={null}
    secondary={false}
    submit={false}
  >
    Sort
  </BpkButton>
  <div
    className="ActionBar__alert"
  >
    <BpkButton
      blank={false}
      className={null}
      destructive={false}
      disabled={false}
      featured={false}
      href={null}
      iconOnly={false}
      large={false}
      link={true}
      onClick={null}
      outline={false}
      rel={null}
      secondary={false}
      submit={false}
    >
      <withAlignment(Component) />
       Price Alerts
    </BpkButton>
  </div>
</div>
`;
